src/FuzzyWalls.cpp

```diff
@@ -0,0 +1,218 @@
+/** Copyright (C) 2016 Tim Kuipers - Released under terms of the AGPLv3 License */
+#include "FuzzyWalls.h"
+
+#define NORMAL_LENGTH 10000
+
+namespace cura 
+{
+
+FuzzyWalls::FuzzyWalls(const SliceMeshStorage& mesh)
+: settings(&mesh)
+{
+    if (mesh.getSettingBoolean("fuzz_map_enabled"))
+    {
+        assert(mesh.texture_proximity_processor && "texture_proximity_processor should have been initialized");
+        getAmplitude = [&mesh, this](const unsigned int layer_nr, const Point p)
+        {
+            assert(mesh.texture_proximity_processor && "When fuzz_map_enabled there has to be a texture proximity processor!");
+            TextureProximityProcessor& texture_proximity_processor = *mesh.texture_proximity_processor;
+            float color = texture_proximity_processor.getColor(p, layer_nr, settings.color_usage, 0.0); // TODO change default 0.0
+            coord_t ret = color * settings.max_amplitude;
+            return ret;
+        };
+    }
+    else
+    {
+        getAmplitude = [this](const unsigned int layer_nr, const Point p)
+        {
+            return settings.max_amplitude;
+        };
+    }
+}
+
+Polygons FuzzyWalls::makeFuzzy(const SliceMeshStorage& mesh, const unsigned int layer_nr, const Polygons& in)
+{
+    Polygons results;
+    if (in.size() == 0)
+    {
+        return results;
+    }
+
+    flows.reserve(in.size());
+    for (ConstPolygonRef poly : const_cast<Polygons&>(in))
+    {
+        assert(poly.size() >= 3);
+        // generate points in between p0 and p1
+        PolygonRef result = results.newPoly();
+        flows.emplace_back(); // keep flows aligned with the result
+        flows.back().reserve(poly.size());
+
+        Point p0 = poly[poly.size() - 2];
+        Point p1 = poly.back();
+        for (int p0_idx = poly.size() - 2; p0_idx >= 0; p0_idx--)
+        { // p0 is the last point before p1 which is different from p1
+            p0 = poly[p0_idx];
+        }
+        CarryOver carry_over;
+        carry_over.dist_left_over = (settings.min_dist_between_points + rand() % settings.range_random_point_dist) / 2;
```

Comment:
The `rand` was already present for the old fuzzy skin. I don't think it's that important to have executions be the same. We can introduce that whenever we get to the point we actually need it.

For now it can be seen as a feature that the random Z seam changes with every slice...